File: backend/app/schemas/provider_connection.py
from pydantic import BaseModel, Field


class ProviderConnectionUpsert(BaseModel):
    api_key: str = Field(
        min_length=8,
        max_length=4096,
    )


class ProviderConnectionResponse(BaseModel):
    provider: str
    name: str
    connected: bool
    key_last_four: str | None
    default_model: str
    models: list[str]
    status: str


class ProviderConnectionTestResponse(BaseModel):
    ok: bool
    message: str

File: frontend/src/features/providerConnections/types/providerConnection.ts
export interface ProviderConnection {
  provider: string;
  name: string;
  connected: boolean;
  key_last_four: string | null;
  default_model: string;
  models: string[];
  status: "connected" | "not_connected";
}

export interface ProviderConnectionUpsert {
  api_key: string;
}

export interface ProviderConnectionTestResponse {
  ok: boolean;
  message: string;
}
